core/amber/src/test/scala/edu/uci/ics/texera/workflow/common/storage/FileDocumentSpec.scala

```diff
@@ -101,11 +101,9 @@ class FileDocumentSpec extends AnyFlatSpec with Matchers with BeforeAndAfter {
       .futureValue
   }
 
-  it should "handle concurrent reads and writes safely" in {
-    Future {
-      val contentStream = new ByteArrayInputStream(newContent.getBytes)
-      fileDocument.appendStream(contentStream)
-    }
+  it should "handle concurrent reads safely" in {
+    val contentStream = new ByteArrayInputStream(newContent.getBytes)
+    fileDocument.appendStream(contentStream)
```

Comment:
I don't get the purpose of this change. Can you elaborate?

Reply:
This change essentially removed `Future {}` wrapped around the `appendStream`. So after line 106, the write is done, and the reader can start to concurrently read from the same file content. 

Before the change, the `appendStream` may happen after one reader reads the file, and that reader will read the old content, which introduces the error.

Reply:
This change essentially removed `Future {}` wrapped around the `appendStream`. So after line 106, the write is done, and the reader can start to concurrently read from the same file content. 

Before the change, the `appendStream` may happen after one reader reads the file, and that reader will read the old content, which introduces the error.

Reply:
I thought you are testing for concurrency. Removing the future will make it synchronized. Then what's the point of having this test case?

Reply:
I only removed the `Future {}` around the appendStream. Meaning that the writer will write the content beforehand sychronously. And the reader will read concurrently.

The purpose of this test is to see if the reader can read the same file content thread-safely. 

Before the change, the writer is also writing the file concurrently. Then what could happen is(the setting is 5 reader, 1 writer):
either:
- reader r1 read file, found that the content is not updated => through errors
- the test failed

or:
- writer append some content to the file
- reader r1 read file, check pass
- r2 read file, check pass
- r3 ...
- r4 ...
- r5 ...
the test passed

Reply:
ok. based on the explanations, we are only testing a subset of concurrent cases. We can merge this PR and fix other untested cases later.